### PR TITLE
Add new Display Name page

### PR DESCRIPTION
**Pull request by Beamanator**
Currently holding on implementation of the `UpdateDisplayName` command - cc @dangrous 

Command ~being~ created here: https://github.com/Expensify/Web-Expensify/pull/35315

### Details

This is the new Display Name page that comes from the N7 Free Plan Account Settings doc. This PR just implements the page, which you'll need to manually navigate to for now. Once all of these pages are built (there's ~ 5 more), we'll hook them all up in a modified Profile page!

### Fixed Issues
$ https://github.com/Expensify/App/issues/11572

### Tests
1. Navigate to http://localhost:8080/settings/profile/display-name
    - Note: You have to do this manually for now, there's no button to navigate you to this new page yet :D
2. Add first & last names that are > 50 characters
3. Click "Save", then verify you see this error:
    - Screen Shot 2022-11-01 at 1 42 57 PM
4. Change the first & last names to something < 50 characters each
5. Click "Save", verify you got navigated back to the Profile page
6. Next, log out & log in, go back to that page, make sure the updates are still there

#### Offline test

1. Navigate to http://localhost:8080/settings/profile/display-name in two devices (or a normal browser & an incognito browser)
2. Go offline on one device, update your name, click "Save"
3. Verify your display name updates everywhere in that device
4. Go back online
5. Verify your display name updates on the second device

- [x] Verify that no errors appear in the JS console

### QA Steps
1. Navigate to http://staging.new.expensify.com/settings/profile/display-name
    - Note: You have to do this manually for now, there's no button to navigate you to this new page yet :D
2. Add first & last names that are > 50 characters
3. Click "Save", then verify you see this error:
    - Screen Shot 2022-11-01 at 1 42 57 PM
4. Change the first & last names to something < 50 characters each
5. Click "Save", verify you got navigated back to the Profile page
6. Next, log out & log in, go back to that page, make sure the updates are still there

#### Offline test

1. Navigate to http://staging.new.expensify.com/settings/profile/display-name in two devices (or a normal browser & an incognito browser)
2. Go offline on one device, update your name, click "Save"
3. Verify your display name updates everywhere in that device
4. Go back online
5. Verify your display name updates on the second device

- [ ] Verify that no errors appear in the JS console

#### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

#### Web

Screen Shot 2022-11-01 at 1 42 57 PM

https://user-images.githubusercontent.com/3885503/199226828-30d4e929-2c3f-4220-a8d1-e45f960e04d2.mov

#### Mobile Web - Chrome

Couldn't get Android chrome working :'( Asking for C+ / reviewer for assistance

#### Mobile Web - Safari

Screen Shot 2022-11-01 at 1 58 40 PM

https://user-images.githubusercontent.com/3885503/199228270-198bbe24-e04a-446c-9259-5698ceec5a3d.mov

#### Desktop

Can't test yet, b/c page isn't hooked up to a button & can't manually navigate in desktop

#### iOS

Can't test yet, b/c page isn't hooked up to a button & can't manually navigate in native

#### Android

Can't test yet, b/c page isn't hooked up to a button & can't manually navigate in native

### CODE REVIEW

File: src/languages/es.js [hunk 1]
```diff
@@ -300,6 +300,12 @@ export default {
         offline: 'Desconectado',
         syncing: 'Sincronizando',
     },
+    displayNamePage: {
+        headerTitle: 'Nombre para mostrar',
+        isShownOnProfile: 'Su nombre para mostrar se muestra en su perfil.',
```
Comment (Beamanator) [inline src/languages/es.js:305]: Getting confirmation on these here: https://expensify.slack.com/archives/C21FRDWCV/p1666786645260019

Comment (Beamanator) [reply src/languages/es.js]: Got confirmation

**Comment by Beamanator** [comment]
@eVoloshchak FYI you probably can't test this yet, till the server-side changes make it to Staging

**Comment by eVoloshchak** [comment]
I don't have the bandwidth right now, could you unassign me please?

**Comment by Beamanator** [comment]
Thanks for letting me know @eVoloshchak !

**Comment by Beamanator** [comment]
@stitesExpensify do you have time to test this or do you prefer we assign a new C+? FYI it's not possible for C+ to test in staging **YET** (waiting on a Web-E PR)

**Comment by stitesExpensify** [comment]
I can test it. I'll try to get to it today

**Comment by Beamanator** [comment]
Thanks @stitesExpensify !

**Comment by stitesExpensify** [comment]
I'll do this tomorrow. Looks like web is still in review as well

**Review by stitesExpensify — CHANGES_REQUESTED** [review]
FYI the offline test does not appear to be working (on both android/web, and web/incognito). LMK if I'm doing something wrong


https://user-images.githubusercontent.com/42391420/199812517-9db52048-6f57-4d38-944a-e2bd9d02783c.mp4

**Comment by Beamanator** [comment]
Huh I swear that was working when I tested 😆

**Comment by Beamanator** [comment]
Weird, I just tested (Chrome, normal + incognito) and I saw it working as expected 😡 Video:

https://user-images.githubusercontent.com/3885503/199964566-e4695285-1f5e-4a5c-b10b-915c515c085a.mov

Just to make sure, you did pull `main` in Web-E right?

**Comment by stitesExpensify** [comment]
hmm I could have sworn I did, but I'll try again!

**Comment by Beamanator** [comment]
Merging main so we're testing on latest 👍

**Comment by stitesExpensify** [comment]
Sorry for the delay, I found out that my local pusher was broken which caused it to not work (obviously). Test passed!

**Review by stitesExpensify — APPROVED** [review]
Code looks great! Nice work!

**Comment by Beamanator** [comment]
Removing emergency label, I guess the PR reviewer checklist may not have been filled out when this was merged 😅

**Comment by stitesExpensify** [comment]
I could have sworn I checked all of those boxes 🙃 My bad

**Pull request merged**